include/dynamatic/Dialect/Handshake/HandshakeOps.td

```diff
@@ -609,7 +615,7 @@ def MemoryControllerOp : Handshake_Op<"mem_controller", [
   let hasVerifier = 1;
   let assemblyFormat = [{
     `[` $memRef `:` type($memRef) `]` $memStart ` ` `(` $inputs `)` $ctrlEnd
-    attr-dict `:` functional-type($inputs, $outputs)
+    attr-dict `:` type($memStart) type($memEnd) type($ctrlEnd) functional-type($inputs, $outputs)
```

Comment:
Since the ControlType now has an argument, we can no longer omit these types. However, it is uniquely determined not to have extra bits because we are using SimpleControl, which restricts the ControlType to no extra bits.
If this seems redundant, we can consider designing a custom assembly printer.